src/pages/projectSettings/tabs/EventLogTab/EventLogDiffs.ts

```diff
@@ -0,0 +1,62 @@
+import { diff } from "deep-object-diff";
+import { string } from "utils";
+
+const { omitTypename } = string;
+
+const isObject = (val) => val && typeof val === "object" && !Array.isArray(val);
+
+const addDelimiter = (a: string, b: string) => (a ? `${a}.${b}` : b);
+
+const getDiffProperties = (eventObj: object) => {
+  const paths = (obj = {}, head = "") =>
+    Object.entries(obj).reduce((event, [key, value]) => {
+      const fullPath = addDelimiter(head, key);
+
+      return isObject(value)
+        ? event.concat(paths(value, fullPath))
+        : event.concat(fullPath);
+    }, []);
+
+  return paths(eventObj);
+};
+
+const formatArrayElements = (eventKey: string) =>
+  eventKey.replace(/.[0-9]./g, (x) => `[${x[1]}].`);
+
+const getNestedObject = (nestedObj: object, pathArr: string[]) =>
+  pathArr.reduce(
+    (obj, key) => (obj && obj[key] !== "undefined" ? obj[key] : undefined),
+    nestedObj
+  );
+
+type EventDiffLine = {
+  key: string;
+  before: any;
+  after: any;
+};
+
+export const getEventDiffLines = (
+  before: object,
+  after: object
```

Comment:
We can more strongly type these values. So they accurately describe the objects within.

Reply:
done